drivers/gnss/gnss_publish.c

```diff
@@ -8,29 +8,33 @@
 #include <zephyr/kernel.h>
 #include <zephyr/sys/iterable_sections.h>
 
-static struct k_spinlock lock;
+static K_SEM_DEFINE(semlock, 1, 1);
 
 void gnss_publish_data(const struct device *dev, const struct gnss_data *data)
 {
-	K_SPINLOCK(&lock) {
-		STRUCT_SECTION_FOREACH(gnss_data_callback, callback) {
-			if (callback->dev == NULL || callback->dev == dev) {
-				callback->callback(dev, data);
-			}
+	(void)k_sem_take(&semlock, K_FOREVER);
```

Comment:
A bit late now that this is merged, but why those casts to void?

Reply:
To inform CI that we are ignoring the return value on purpose :) take with K_FOREVER can only return 0, so we don't care to check it :)

Reply:
Yeah, but I mean would the removal of `(void)` provoke build errors? :thinking:

Reply:
@tomi-font this is part of our coding guidelines :)

> 17.7 The value returned by a function having non-void return type shall be used

https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_17_07.c

Reply:
> @tomi-font this is part of our coding guidelines :)
> 
> > 17.7 The value returned by a function having non-void return type shall be used
> 
> https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_17_07.c

Are we following MISRA C to the letter? :open_mouth: